docs/docs/aleph_zero/sdk.md

```diff
@@ -0,0 +1,386 @@
+
+```typescript
+const main = async () => {
+  // --snip--
+
+  // load invariant contract
+  const invariant = await Invariant.load(api, Network.Main, INVARIANT_ADDRESS)
+
+  // load token contracts
+  const token0 = await PSP22.load(api, Network.Main, TOKEN0_ADDRESS) // token we want to give
+  const token1 = await PSP22.load(api, Network.Main, TOKEN1_ADDRESS) // token we want to receive
+
+  // set fee tier and pool key, make sure that pool with specified parameters exists
+  const feeTier = newFeeTier(toFee(1n, 2n), 1n) // fee: 0.01 = 1%, tick spacing: 1
+  const poolKey = newPoolKey(
+    token0.contract.address.toString(),
+    token1.contract.address.toString(),
+    feeTier
+  )
+
+  // make sure that you swap in correct direction,
+  // here we check if token0 (token we want to give) is tokenX
+  let xToY = isTokenX(token0.contract.address.toString(), token1.contract.address.toString())
+}
+
+main()
+```
+
+### Perform swap
+
+:::info How to calculate input amount
+In order to calculate input amount, we have to multiply actual token amount you want to swap times 10 to the power of decimal.
+Let's say some token has decimal of 12 and we want to swap 6 actual tokens. Here is how we can calculate input amount:
+6 \* 10^12 = 6000000000000.
+:::
+
+```typescript
+const main = async () => {
+  // --snip--
+
+  // here we want to swap 6 token0
+  // token0 has 12 decimals so we need to multiply it by 10^12
+  const amount = 6n * 10n ** 12n
+
+  // approve token0 transfer
+  await token0.approve(account, invariant.contract.address.toString(), amount)
+
+  // slippage is a price change you are willing to accept,
+  // for examples if current price is 1 and your slippage is 1%, then price limit will be 1.01
+  const allowedSlippage = toPercentage(1n, 3n) // 0.001 = 0.1%
+
+  // swap message only accepts price limit so we have to calculate it in order to use slippage,
+  // here we are passing current price and allowed slippage to calculate price limit
+  const pool = await invariant.getPool(
+    account,
+    token0.contract.address.toString(),
+    token1.contract.address.toString(),
+    feeTier
+  )
+  const sqrtPriceLimit = calculateSqrtPriceAfterSlippage(pool.sqrtPrice, allowedSlippage, xToY)
+
+  const result = await invariant.swap(account, poolKey, xToY, amount, true, priceLimit)
+  console.log(result.hash) // print transaction hash
+}
+
+main()
+```
+
+### Query state
+
+```typescript
+const main = async () => {
+  // --snip--
+
+  // get specific user address
+  const userAddress = account.address.toString()
+
+  // retrieve positions of that user
+  const positions = invariant.getPositions(account, userAddress)
+
+  const fees = new Map()
+
+  // loop through user's positions
+  positions.map(position => {
+    // get token x and y addresses
+    const tokenX = position.poolKey.tokenX
+    const tokenY = position.poolKey.tokenY
+
+    // add unclaimed fees to a map
+    map.set(tokenX, (map.get(tokenX) || 0) + position.tokensOwedX)
```

Comment:
`position.tokensOwed` isn't the token inside the position. It's an unclaimed fee, but the state isn't up to date